includes/uploader.js

```diff
@@ -121,17 +135,64 @@ function check_file_content(isValidFile) {
 
         reader.readAsText(file);
     }
+
 }
 
 
 
-// Check file content 
+function check_columns_names(isValidContent) {
+    if (isValidContent) {
+        // Define the names accepted for the columns
+        const columns_names_reference_list = [
+            "Statement Type",
+            "Attribute",
+            "Deontic",
+            "Aim",
+            "Direct Object",
+            "Type of Direct Object",
+            "Indirect Object",
+            "Type of Indirect Object",
+            "Activation Condition",
+            "Execution constraint",
+            "Or else\r"
+
+        ];
+
+        // Array to store unrecognized column names
+        const unrecognizedColumns = [];
+
+        // Iterate through columnNames and check correspondence
+        columnNames.forEach(function (name, index) {

```

Comment:
Instead of reading a global `columnNames`, can we pass the column names as an argument to the function? That would make the function more self-contained and prevent any issues with unforeseen changes to the global variable.